File: packages/hmssdk_flutter/android/src/main/kotlin/live/hms/hmssdk_flutter/HMSResultExtension.kt
package live.hms.hmssdk_flutter

class HMSResultExtension {

    companion object {
        fun toDictionary(success: Boolean, data: Any? = null): HashMap<String, Any?> {
            val hashMap = HashMap<String, Any?>()
            hashMap["success"] = success
            hashMap["data"] = data
            return hashMap
        }
    }
}

File: packages/hmssdk_flutter/ios/Classes/Models/HMSResultExtension.swift
//
//  HMSResultExtension.swift
//  hmssdk_flutter
//
//  Created by Pushpam on 27/03/23.
//

import Foundation

class HMSResultExtension {

    static func toDictionary(_ success: Bool, _ data: Any? = nil) -> [String: Any?] {

        /**
            We  add `data`  field in the map only when it is non null
             otherwise the application might crash.
         */
        var dict: [String: Any] = ["success": success]
        if let data = data {
            dict["data"] = data
        }
        return dict
    }
}
